### PR TITLE
Navbar 

### PR DESCRIPTION
**Pull request by olaouso**
Screenshot (235)
This code for the Navbar part, it #includes the logo, website pages which are (Home, Posts, donate, about us, contact us) in addition to the search part then three buttons (search, login, sign up).
Also, the navbar included a dropdown list to change the language between EN, AR, or TR.
#14

### CODE REVIEW

File: src/components/Navbar/index.css [hunk 1]
```diff
@@ -0,0 +1,62 @@
+@import url("https://fonts.googleapis.com/css?family=Acme&display=swap");
+
+.NavC {
```
Comment (Ammar-64) [inline src/components/Navbar/index.css]: what is NavC, classNames should be meaningful even if you have to make them long

File: src/components/Navbar/index.css [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  font-family: "Acme" !important;
+  color: #590995 !important;
+}
+.SignUpButton {
```
Comment (Ammar-64) [inline src/components/Navbar/index.css]: there is to much repetition in both buttons you have to follow DRY rule (don't repeat yourself) so you can do one class for everything mutual in both buttons and you can add the unique things to different classes

File: src/components/Navbar/index.css [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+}
+.search {
+  font-family: "Acme" !important;
+  color: #590995 !important;
+  margin-right: 7% !important;
+  /* 
+border: 1px solid #590995 !important; */
+}
+.LogInButton {
+  font-family: "Acme" !important;
```
Comment (Ammar-64) [inline src/components/Navbar/index.css:8]: your using font-family in every single class while you can use it in all of them at the same time and DRY

File: src/components/Navbar/index.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  NavDropdown,
+  Form,
+  FormControl,
+  Button,
+} from "react-bootstrap";
+
+export default function SiteNavbar() {
+  return (
+    <Navbar expand="lg" className="NavC">
+      <Navbar.Brand href="#home">
```
Comment (Ammar-64) [inline src/components/Navbar/index.jsx]: you should use link not href

File: src/components/Navbar/index.jsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        <Form inline className="search">
+          <FormControl type="text" placeholder="Search here" />
+        </Form>
+
+        <Button className="LogInButton">Search</Button>
+        <Button className="LogInButton">Log in</Button>
+        <Button className="SignUpButton">Sign up</Button>
+
+        <NavDropdown title="EN" id="basic-nav-dropdown" className="NavC">
+          <NavDropdown.Item href="#action/3.1">EN</NavDropdown.Item>
```
Comment (Ammar-64) [inline src/components/Navbar/index.jsx]: what is action/3.1 means

File: src/components/Navbar/index.css [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  margin-right: 0.25%;
+  margin-left: 0.5%;
+  border: 2px solid #590995 !important;
+  box-shadow: 0 12px 16px 0 rgba(0, 0, 0, 0.24),
+    0 17px 50px 0 rgba(0, 0, 0, 0.19);
+  background-color: #ffffff !important;
+  text-align: center !important;
+  padding: 0 !important;
+}
+.LogInButton:visited {
```
Comment (Ammar-64) [inline src/components/Navbar/index.css]: classNames should be camelCases like logInButton

File: src/App.jsx [hunk 7]
```diff
@@ -1,9 +1,10 @@
 import React from "react";
 import Home from "./containers/Home";
-
+import SiteNavbar from "./components/Navbar";
```
Comment (Ammar-64) [inline src/App.jsx:6]: keep a line space between your imports and your components

**Review by Ammar-64 — CHANGES_REQUESTED** [review]

**Review by Luey-A — APPROVED** [review]

**Review by Luey-A — CHANGES_REQUESTED** [review]
I request all the changes that were requested by Ammar.

File: src/components/Navbar/index.css (hunk 1 above)
Comment (olaouso) [reply src/components/Navbar/index.css]: replaced with NavMain

File: src/components/Navbar/index.jsx (hunk 4 above)
Comment (olaouso) [reply src/components/Navbar/index.jsx]: link didn't work in bootstrap

File: src/components/Navbar/index.jsx (hunk 5 above)
Comment (olaouso) [reply src/components/Navbar/index.jsx]: revised

File: src/components/Navbar/index.css [hunk 8]
```diff
@@ -0,0 +1,60 @@
+@import url("https://fonts.googleapis.com/css?family=Acme&display=swap");
+
+.NavMainClass {
```
Comment (Ammar-64) [inline src/components/Navbar/index.css]: classNames should be camelCasses

File: src/components/Navbar/index.css (hunk 3 above)
Comment (Ammar-64) [reply src/components/Navbar/index.css:8]: This is not resolved yet

**Review by Ammar-64 — COMMENTED** [review]
Please resolve the duplication in css, and class names

File: src/components/Navbar/index.css [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  border: 0.2em solid #590995 !important;
+  text-align: center !important;
+  padding: 0 !important;
+}
+.search {
+  font-family: "Acme" !important;
+  color: #590995 !important;
+  margin-right: 7% !important;
+  /* 
+border: 1px solid #590995 !important; */
```
Comment (Luey-A) [inline src/components/Navbar/index.css]: comments should be removed before the pull request

File: src/components/Navbar/__snapshots__/index.test.jsx.snap [hunk 10]
```diff
@@ -0,0 +1,132 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (Luey-A) [inline src/components/Navbar/__snapshots__/index.test.jsx.snap:1]: comments should be removed before the pull request

File: src/App.jsx [hunk 11]
```diff
@@ -1,9 +1,20 @@
...
 function App() {
   return (
     <div>
+      <Router>
+        {/* <Route exact path="/" component={Home} /> */}
+        {/* <Route path="/Posts" component={Posts} />
+      <Route path="/Donate" component={Donate} />
+      <Route path="/Aboutus" component={Aboutus} />
+      <Route path="/Contactus" component={Contactus} /> */}
```
Comment (Luey-A) [inline src/App.jsx:15]: comments should be removed before the pull request

File: src/components/Navbar/index.css [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+  top: 33px;
+}
+.search {
+  font-family: "Acme" !important;
+  color: #590995 !important;
+  height: 2em;
+  width: 5em;
+  left: 5em;
+  top: 5em;
+
```
Comment (Luey-A) [inline src/components/Navbar/index.css]: Do you need this empty line?

File: src/locales/en.json [hunk 13]
```diff
@@ -1,8 +1,16 @@
...
-  "links": {
-    "home": "",
-    "about": ""
-  },
   "home": {},
-  "about": {}
+  "about": {},
+
+  "navbarsection": {
+    "Home": "Home",
```
Comment (Luey-A) [inline src/locales/en.json:6]: the above words start with a small letter when these ones starts with a big letter! for example: home in the links and Home in the navbarsection.

**Review by knargocer — APPROVED** [review]

File: src/components/Navbar/index.css (hunk 6 above)
Comment (olaouso) [reply src/components/Navbar/index.css]: done

File: src/App.jsx (hunk 7 above)
Comment (olaouso) [reply src/App.jsx:6]: done

File: src/components/Navbar/index.css (hunk 2 above)
Comment (olaouso) [reply src/components/Navbar/index.css]: Done

File: src/components/Navbar/index.css (hunk 3 above)
Comment (olaouso) [reply src/components/Navbar/index.css:8]: done

File: src/locales/en.json [hunk 14]
```diff
@@ -1,8 +1,16 @@
...
+    "Posts": "Posts",
+    "Donate": "Donate",
+    "AboutUs": "About us",
+    "ContactUs": "Contact us",
+    "SearchHere": "Search here",
+    "Search": "Search",
+    "Login": "Login",
```
Comment (Ammar-64) [inline src/locales/en.json]: LogIn should be written like this

**Review by Ammar-64 — CHANGES_REQUESTED** [review]
There is one comment resolve it, please

File: src/locales/en.json (hunk 14 above)
Comment (olaouso) [reply src/locales/en.json]: resolved

File: src/components/Navbar/index.jsx [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+      <Navbar.Brand href="/">
+        <img src="../logo.png" alt="logo" width="75em"></img>
+      </Navbar.Brand>
+      <Navbar.Toggle aria-controls="basic-navbar-nav" />
+      <Navbar.Collapse id="basic-navbar-nav">
+        <Nav className="mr-auto">
+          <Nav.Link href="/" className="navMainClass">
+            {t("navbarsection.Home")}
+          </Nav.Link>
+          <Nav.Link href="#Posts" className="navMainClass">
```
Comment (Ammar-64) [inline src/components/Navbar/index.jsx]: The links should start with a ”/“ instead of “#”

**Review by Ammar-64 — CHANGES_REQUESTED** [review]
Fix the links in the navbar please

**Review by Ammar-64 — APPROVED** [review]

**Comment by olaouso** [comment]
@Luey-A PTAL and approve so the branch can be merged

**Review by Luey-A — APPROVED** [review]

**Pull request merged**